src/main/java/LinkedList.java

```diff
@@ -65,36 +61,42 @@ public void reverse() {
 		head = prev;
 	}
 	
+	public static boolean isCyclical(LinkedList listA) {
```

Comment:
This works; next assignment: rewrite isCyclical so that it doesn't take extra memory (ie you're not using a HashSet to keep track of anything).

Reply:
One way to do it, off the top of my head, is by keeping track of size as state and rewriting isCyclical to loop through the nodes and returning true if the loop iterations exceeds the size